modules/storage-aws-s3/s3/s3.go

```diff
@@ -102,8 +111,56 @@ func (s *s3) StoreSnapshot(ctx context.Context, snapshot snapshots.Snapshot) err
 	return nil
 }
 
+func (s *s3) makeFilePath(dataPath, relPath string) string {
+	return fmt.Sprintf("%s/%s", dataPath, relPath)
+}
+
+func (s *s3) makeObjectName(snapshotId, srcRelPath string) string {
+	return fmt.Sprintf("%s/%s", snapshotId, srcRelPath)
+}
```

Comment:
Maybe consider removing these methods and replacing their usage with `path.Join` to simplify things

Reply:
My concern here is less about simplicity but reliability.  If anyone changes the path later, they will have to change it in multiple places.  It's possible to miss one of those places.  By putting the (simple) logic in one place, rather than spreading it out in the file, we can change the logic in one place and it will be used correctly everywhere.

Reply:
agreed, lets kept explicit !